File: app/src/main/java/com/mgn/bingenovelreader/models/WebPage.java
package com.mgn.bingenovelreader.models;

public class WebPage {

    public WebPage() {
        //Empty Constructor
    }

    public WebPage(String url, String title, String pageData) {
        this.url = url;
        this.title = title;
        this.pageData = pageData;
    }


    private long id;
    private String url;
    private String title;
    private String fileName;
    private long fictionId;
    public String pageData;

    public void setId(long id) {
        this.id = id;
    }

    public long getId() {
        return this.id;
    }

    public String getUrl() {
        return this.url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getTitle() {
        return this.title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getFileName() {
        return this.fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public long getFictionId() {
        return this.fictionId;
    }

    public void setFictionId(long fictionId) {
        this.fictionId = fictionId;
    }

}

File: app/src/main/java/com/mgn/bingenovelreader/models/WebPage.kt
package com.mgn.bingenovelreader.models

class WebPage {

    constructor() {
        //Empty Constructor
    }

    constructor(url: String, chapter: String) : super() {
        this.url = url
        this.chapter = chapter
    }

    constructor(url: String, chapter: String, pageData: String) : this(url, chapter) {
        this.pageData = pageData
    }

    var id: Long? = -1
    var url: String? = null
    var redirectedUrl: String? = null
    var title: String? = null
    var chapter: String? = null
    var filePath: String? = null
    var novelId: Long = 0
    var pageData: String? = null

}
